### DISCUSSION TITLE
Track number into tags

### DISCUSSION THREAD

**Discussion by Mihokas99** [section83/MacYTDL#78] [Ideas]
I tried download one playlist (audio only) and it worked great. When looking at mp3 tag information they look fine as well. Only crucial thing missing was track number. I looked through settings and did not find anything that would help. So maybe this is something that could be added in the future?

**Comment by section83** [comment]
Hello Mihokas99, thank you for making contact.

MacYTDL doesn't collect track or playlist number data at present.  However, there are ways to get metadata.  There is an "Add metadata" setting in "Settings" which sometimes works. I find that it's not very good at getting metadata from YouTube.

You could also try the yt-dlp settings such as: --write-info-json ; --write-playlist-metafiles. You can add those to MacYTDL via the "Use custom settings" setting.  Also some post-processors can collect metadata (perhaps including FFmpeg). I've not tried any of those so, can't help with using them.  Post-processor commands can be added to MacYTDL also via the "Use custom settings" setting. The [YT-DLP readme page](https://github.com/yt-dlp/yt-dlp/blob/master/README.md) has guidance on those settings.

I hope to improve playlist downloading in a future version of MacYTDL.  That will include providing a dialog to select playlist items for download. It will probably need a playlist number to do that.

Cheers.
